django/db/models/fields/related.py

```diff
@@ -1586,7 +1587,20 @@ class ForeignObject(RelatedField):
 
     def __init__(self, to, from_fields, to_fields, rel=None, related_name=None,
             related_query_name=None, limit_choices_to=None, parent_link=False,
-            on_delete=CASCADE, swappable=True, **kwargs):
+            on_delete=None, swappable=True, **kwargs):
```

Comment:
Well, we _could_ make `on_delete` an actually-required arg to `ForeignObject` right now, and move it even before `from_fields` and `to_fields`, but that would require duplicating the deprecation warning in both `ForeignKey` and `OneToOneField`.